File: movie-catalog-master/src/main/java/hu/elte/alkfejl/moviecatalog/model/Director.java
package hu.elte.alkfejl.moviecatalog.model;

import java.util.HashSet;
import java.util.Objects;
import java.util.Set;
import lombok.Data;

import javax.persistence.*;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "DIRECTOR")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Director extends BaseEntity {

  
    @Column(name = "director_name", nullable = false)
    private String name;

    @Column(nullable = false)
    private String nationality;

    @Column(nullable = false, columnDefinition = "text")
    private String bio;

    @ManyToMany(mappedBy = "directors", cascade = CascadeType.ALL)
    private Set<Movie> movies = new HashSet<>();
   

}   

File: ng-moviecatalog/src/app/classes/director.ts
export class Director {
  id: number;
  name: string;
  nationality: string;
  bio: string;

  public constructor(id?: number,
                     name?: string,
                     nationality?: string,
                     bio?: string) {
    this.id = id;
    this.name = name;
    this.nationality = nationality;
    this.bio = bio;
  }
}
